Fix for the latest SNP kernel (for versions 6.8 and above)

LakshmiSaiHarika opened the pull request:
Updated snp setup-host and launch-guest feature to launch latest kernel SNP (6.9) enabled Guest

LakshmiSaiHarika committed:
snp.sh: Standardize SNP host and guest kernel retrieval

Added the SNP host kernel retrieval to grab SNP host kernel version from AMDSEV config file across different linux distros for setting default grub menu.
Further, SNP host kernel retrieval fixes no such directory error in ubuntu because of missing /linux-image folder under AMDSEV directory.

Added the SNP guest kernel retrieval to standardize guest file location across various linux distros,
because AMDSEV does not copy bzImage into boot folder in some linux distros (Redhat, Fedora) after AMDSEV package build step

Changed snp-release directory to the latest snp-release directory
Signed-off-by: Harika <lnittala@amd.com>

LakshmiSaiHarika committed:
snp.sh: Image variable update

Signed-off-by: Harika <lnittala@amd.com>

LakshmiSaiHarika committed:
snp.sh: ovmf and bios update for latest SNP kernel fix

Based on the latest AMDSEV documentation,
Deleted ovmf/ directory prior to build step for ovmf clean re-initialization
Added qemu command line option for bios
Removed qemu command lind for pflash as SNP no longer supports using pflash unit=0 for loading the bios, and
instead relies on -bios parameter

Signed-off-by: Harika <lnittala@amd.com>

ryansavino commented on tools/snp.sh:
```diff
@@ -271,10 +271,18 @@ install_dependencies() {
+  # Retrieve SNP host kernel from the host kernel config file
+  local host_kernel_config="${SETUP_WORKING_DIR}/AMDSEV/linux/host/.config"
+  local kernel_version=$(cat ${host_kernel_config} | grep "^# Linux/.*Kernel Configuration$" | awk -F ' ' '{print $3}')
+  local kernel_type=$(cat ${host_kernel_config} | grep "CONFIG_LOCALVERSION" | grep -v "^#"|awk -F '="' '{print $2}'|tr -d '"')
```
spacing

ryansavino replied:
also, name this `kernel_hash`

ryansavino commented on tools/snp.sh:
```diff
@@ -490,11 +498,22 @@ build_guest_initrd() {
+  # Retrieve SNP guest kernel from the guest kernel config file
+  local guest_kernel_config="${SETUP_WORKING_DIR}/AMDSEV/linux/guest/.config"
+  local kernel_version=$(cat ${guest_kernel_config} | grep 'Linux/' | awk -F ' ' '{print $3}')
+  local kernel_type=$(cat ${guest_kernel_config} | grep "CONFIG_LOCALVERSION" | grep -v "^#"|awk -F '="' '{print $2}'|tr -d '"')
```
`kernel_hash` and spacing

ryansavino commented on tools/snp.sh:
```diff
@@ -271,10 +271,18 @@ install_dependencies() {
 }
 
+get_host_kernel_version() {
+  # Retrieve SNP host kernel from the host kernel config file
```
move comment to above method declaration, and mention the hash as well

ryansavino commented on tools/snp.sh:
```diff
@@ -490,11 +498,22 @@ build_guest_initrd() {
 }
 
+get_guest_kernel_version() {
+  # Retrieve SNP guest kernel from the guest kernel config file
```
move comment to above method declaration, and mention the hash as well

ryansavino commented on tools/snp.sh:
```diff
@@ -643,8 +662,20 @@ build_and_install_amdsev() {
+  local guest_kernel_version=$(get_guest_kernel_version)
+
+  # To standardize guest kernel file location across different linux distributions.
+  local bzImage_file=$(find ${SETUP_WORKING_DIR}/AMDSEV/linux/guest -name "bzImage"| head -1)
```
Spacing. Be consistent with comment punctuation. For single statements elsewhere in comments, you don't use a period.

ryansavino commented on tools/snp.sh:
```diff
@@ -643,8 +662,20 @@ build_and_install_amdsev() {
+
+  # To standardize guest kernel file location across different linux distributions.
+  local bzImage_file=$(find ${SETUP_WORKING_DIR}/AMDSEV/linux/guest -name "bzImage"| head -1)
+  local guest_kernel_bin="${SETUP_WORKING_DIR}/AMDSEV/linux/guest/vmlinuz-$guest_kernel_version"
```
Braces on variable.

ryansavino commented on tools/snp.sh:
```diff
@@ -643,8 +662,20 @@ build_and_install_amdsev() {
+  # Guest kernel binary is not present inside guest directory for some Linux Distributions like RH and Fedora,
+  # because AMDSEV does not copy guest kernel file inside guest directory during SNP package build process in RH, fedora
+  # so, copying bzImage file into guest kernel binary file if vmlinuz is absent inside the guest directory
+  [ -f $guest_kernel_bin ] || cp -v $bzImage_file $guest_kernel_bin
```
Braces.

ryansavino reviewed (changes requested):
Looks much cleaner. Please fix requested changes.

LakshmiSaiHarika committed:
snp.sh: Increased timer for guest complete shutdown

Increased timer for guest complete shutdown to 10 seconds to avoid qemu failed to get write lock on guest image issue

Signed-off-by: Harika <lnittala@amd.com>
